modules/webrequest-pipeline/sources/webrequest-context.es

```diff
@@ -65,8 +65,8 @@ export default class WebRequestContext {
 
     // Cliqz-specific extensions to webRequest details
     context.page = page;
-    context.tabUrl = context.tabUrl || (page && page.getTabUrl());
-    context.frameUrl = context.frameUrl || (page && page.getFrameUrl(context));
+    context.tabUrl = context.tabUrl || (page && page.getTabUrl()) || context.url;
+    context.frameUrl = context.frameUrl || (page && page.getFrameUrl(context)) || context.url;
```

Comment:
why this needs a change?

Reply:
I'm not sure if it is sound to fill it with the url. But the reason that I added it was that there are multiple places where the adblocker assumes that it is present, for instance here: https://github.com/ghostery/common/blob/319cc4fd3c99db6733010b051663093c2959fa68/modules/adblocker/sources/adblocker.es#L235

And later it would throw in the whitelist check:
https://github.com/ghostery/common/blob/319cc4fd3c99db6733010b051663093c2959fa68/modules/adblocker/sources/adblocker.es#L255

Reply:
Looks like this here is the main problem still. Filling it in like this is risky. Potentially we could apply it only if it is a request from a service worker; but even then, it is not clear which URL to fill in.